core/src/main/java/io/smallrye/context/CompletableFutureWrapper.java

```diff
@@ -22,9 +22,12 @@
      * another CF, so we have different behaviour
      */
     protected final Executor executor;
-    private final boolean minimal;
+    protected final int flags;
 
-    public CompletableFutureWrapper(SmallRyeThreadContext context, CompletableFuture<T> f, Executor executor, boolean minimal) {
+    public final static int FLAG_MINIMAL = 1 << 0;
+    public final static int FLAG_DEPENDENT = 1 << 1;
```

Comment:
These are really hard to deciphe. Was there any special reason why ordinary conditions wouldn't work?

Reply:
I was at the point where I had two `boolean` parameters to the constructor, and that's where normally we switch to enums, but I figured creating two enums `Something.MINIMAL/NORMAL` `SomethingElse.DEPENDENT/WRAPPER` didn't feel a lot better than flags.
If you feel strongly in favour of two enums, or perhaps using an `EnumSet` in a single enum, I can change it, but two `boolean` in a row are a good reason to find an alternative.

Reply:
I see. I don't really have a strong argument to convince you, it's just that for me the readability of bitwise operations for conditions was always really low. If you feel like it's the easiest way to do it, keep it :-)

Reply:
I think it's fine.